src/common/image.c

```diff
@@ -1148,6 +1148,39 @@ int32_t dt_image_duplicate(const dt_imgid_t imgid)
   return dt_image_duplicate_with_version(imgid, -1);
 }
 
+static int32_t _image_get_possible_version(const dt_imgid_t imgid,
+                                            const int32_t max_version)
+{
+  int32_t safe_max_version = 0;
+  char imgpath[PATH_MAX] = { 0 };
+  char versionpath[PATH_MAX] = { 0 };
+
+  gboolean from_cache = FALSE;
+  dt_image_full_path(imgid, imgpath, sizeof(imgpath), &from_cache);
+
+  for(int32_t version = MAX(1, max_version); version < 100; version++)
```

Comment:
The sidecar files can use up to four digits, so version numbers up to 9999 are possible (though of course extremely unlikely).

Reply:
Oops, I misread the code elsewhere, it's just at least two decimal digits.

Reply:
fixed with force push